### DISCUSSION TITLE
Is it secure to use Supabase Auth in fully front-end projects?

### DISCUSSION THREAD

**Discussion by HappyCodingCoding** [supabase/supabase#29244] [Questions] [answered]
Hey, guys, I have a question to consult with you.

Since all the tokens are stored in local storage and can be accessed by JavaScript, they are easily stolen by someone. So, is it secure to use Supabase Auth in fully front-end projects? Thank you!

**Comment by GaryAustin1** [comment, marked as answer]
It all depends on the risk of access to a user's data and the desire of someone to attack it.  A banking app probably should not use jwt in local storage.  A gaming app where only a user's data get hacked, not as big a concern.

You will see a range of answers on the internet.  Supabase gives you the option to do a browser only app or to doing everything serverside and in between.  Firebase also uses jwts in localstorage.  Here was one article on localstorage, there are many on both sides: https://pragmaticwebsecurity.com/articles/oauthoidc/localstorage-xss.html

Someone has to want to attack your specific app.  If they can get to local storage thru an XSS attack (bad browser extensions or you install bad 3rd party code) then they can also see anything the user is doing (think typing password) or make calls while the user is logged in from your app anyway.

But to steal the jwt and refresh token they have to steal it and use it before the jwt expires and before the refresh token is used on the app itself as it can only be used once.  The shorter the time period for jwt expire, the safer it is .  5 minutes is about as short as you want to go, default is 1 hour.

**Reply by HappyCodingCoding** [reply]
Thank you very much for your help that not only has answered my questions but also has taught me the new knowledge of how to use jwts in a browser only app!
